sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -52,8 +52,11 @@ def test_too_many_components(algorithm, X_sparse):
     n_features = X_sparse.shape[1]
     for n_components in (n_features, n_features + 1):
         tsvd = TruncatedSVD(n_components=n_components, algorithm=algorithm)
-        with pytest.raises(ValueError):
+        if n_components == n_features and algorithm == "randomized":
             tsvd.fit(X_sparse)
+        else:
+            with pytest.raises(ValueError):
+                tsvd.fit(X_sparse)
```

Comment:
Since this test is call "test_too_many_components" I think it is geared toward errors. We can refactor the test as follows:

```python
@pytest.mark.parametrize(
    "algorithm, n_components",
    [
        ("arpack", 55),
        ("arpack", 56),
        ("randomized", 56),
    ],
)
def test_too_many_components(X_sparse, algorithm, n_components):
    tsvd = TruncatedSVD(n_components=n_components, algorithm=algorithm)
    with pytest.raises(ValueError):
        tsvd.fit(X_sparse)
```

Reply:
Done.  Thank you for the feedback!